### PR TITLE
Added Winston & Morgan logging

### PR DESCRIPTION
**Pull request by PatrickShaw**
We need some sort of logging functionality, so I added some.
image

Also added tests because that's a thing we can do now :D

### CODE REVIEW

File: src/backend/logger.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+// Adds a timestamp to the logger information
+const swaggerPlatformTimestamp = format(info => {
+  info.timestamp = moment(new Date()).format('YYYY-MM-DD HH:mm:ss');
+  return info;
+});
+
+// Aligns all the information before the message section of the log
+const swaggerPlatformAlign = format(info => {
+  info.levelPadding = ' '.repeat(9 - info.level.length);
```
Comment (calebj0seph) [inline src/backend/logger.tsx]: Might be worth moving the 9 out into a constant?

File: src/backend/logger.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        swaggerPlatformPrinter,
+      ),
+    }),
+    // Note that we don't want color codes in our file logs (looks incredibly confusing and ugly)
+    new transports.File({
+      filename: 'swagger-platform.error.log',
+      level: 'error',
+      format: swaggerPlatformPrinter,
+    }),
+    new transports.File({
```
Comment (calebj0seph) [inline src/backend/logger.tsx]: Should we document that these files are created somewhere?

File: package.json [hunk 3]
```diff
@@ -66,7 +66,7 @@
     "babel-eslint": "8.2.2",
     "babel-jest": "23.0.0-beta.1",
     "babel-loader": "8.0.0-beta.2",
-    "babel-plugin-module-resolver": "3.1.0",
+    "babel-plugin-module-resolver": "^3.1.1",
```
Comment (calebj0seph) [inline package.json]: Not sure if you meant for a caret to be here.

File: src/backend/logger.tsx (hunk 1 above)
Comment (PatrickShaw) [reply src/backend/logger.tsx]: Yeah alright, no one wants magic numbers.

File: src/backend/logger.tsx (hunk 2 above)
Comment (PatrickShaw) [reply src/backend/logger.tsx]: I personally think it's straight forward enough. Might change the location/get rid of it entirely so it's not worth documenting for now.

**Pull request merged**